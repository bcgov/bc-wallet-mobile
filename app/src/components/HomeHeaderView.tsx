import { useAgent } from '@credo-ts/react-hooks'
import { Screens, Stacks, testIdWithKey, useTheme } from '@hyperledger/aries-bifold-core'
import { RemoteLogger } from '@hyperledger/aries-bifold-remote-logs'
import { useNavigation } from '@react-navigation/native'
import React from 'react'
import { useTranslation } from 'react-i18next'
import { Pressable, StyleSheet, Text, View } from 'react-native'

const HomeHeaderView = () => {
  const { ColorPallet } = useTheme()
  const { t } = useTranslation()
  const { agent } = useAgent()
  const logger = agent?.config.logger as RemoteLogger
  const navigation = useNavigation()

  const styles = StyleSheet.create({
    banner: {
      padding: 10,
      backgroundColor: ColorPallet.notification.errorText,
      justifyContent: 'center',
      alignItems: 'center',
    },
    bannerText: {
      color: ColorPallet.grayscale.white,
      fontSize: 16,
      fontWeight: 'bold',
    },
  })

  const onPressBanner = () => {
    navigation.getParent()?.navigate(Stacks.SettingStack as never, { screen: Screens.Developer } as never)
  }

  return logger?.remoteLoggingEnabled ? (
    <Pressable onPress={onPressBanner} testID={testIdWithKey('SessionIdBanner')}>
      <View style={styles.banner}>
        <Text style={styles.bannerText}>{t('RemoteLogging.Banner', { sessionId: logger.sessionId.toString() })}</Text>
      </View>
    </Pressable>
  ) : null
}

export default HomeHeaderView
