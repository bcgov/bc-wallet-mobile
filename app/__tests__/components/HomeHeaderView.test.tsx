import { render } from '@testing-library/react-native'
import React from 'react'

import HomeHeaderView from '../../src/components/HomeHeaderView'

const mockNavigation = jest.fn()
jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => {
    return mockNavigation
  },
}))

describe('HomeHeaderView Component', () => {
  test('renders correctly', () => {
    const tree = render(<HomeHeaderView />)
    expect(tree).toMatchSnapshot()
  })
})
