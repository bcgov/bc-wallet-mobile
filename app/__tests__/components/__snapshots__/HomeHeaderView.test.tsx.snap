// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeHeaderView Component renders correctly 1`] = `null`;
